### ISSUE TITLE
fioctl crashes when host has no network access

### ISSUE THREAD

**Issue by doanac** [foundriesio/fioctl#74]
Running this command when the host is disconnected from the network:
~~~
(17:38) ~ $ fioctl device list
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x143830d]
goroutine 1 [running]:
github.com/foundriesio/fioctl/client.(*Api).Get(0xc0001f2000, 0xc00010c780, 0x5b, 0xc00016d690, 0x100addb, 0x10)
        /fioctl/client/foundries.go:406 +0x6d
github.com/foundriesio/fioctl/client.(*Api).DeviceListCont(0xc0001f2000, 0xc00010c780, 0x5b, 0x17, 0xc00016d6e8, 0x1)
        /fioctl/client/foundries.go:514 +0x45
github.com/foundriesio/fioctl/client.(*Api).DeviceList(0xc0001f2000, 0xc00016d901, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /fioctl/client/foundries.go:510 +0x2f9
github.com/foundriesio/fioctl/subcommands/devices.doList(0xc0001de780, 0x1a0d110, 0x0, 0x0)
        /fioctl/subcommands/devices/list.go:104 +0xa15
github.com/spf13/cobra.(*Command).execute(0xc0001de780, 0x1a0d110, 0x0, 0x0, 0xc0001de780, 0x1a0d110)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x19d23a0, 0x104460a, 0x1990300, 0xc000054778)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x30b
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/foundriesio/fioctl/cmd.Execute()
        /fioctl/cmd/root.go:37 +0x31
main.main()
        /fioctl/main.go:6 +0x25
~~~

**Comment by doanac** [comment]
@vkhoroz - this has to do with the new `httpLogger` helper. `res` is null in some RawGet failure paths. Here's one approach:
~~~
$ git diff
diff --git a/client/foundries.go b/client/foundries.go
index 8771742..eb6bf49 100644
--- a/client/foundries.go
+++ b/client/foundries.go
@@ -417,7 +417,7 @@ func (a *Api) RawGet(url string, headers *map[string]string) (*http.Response, er
 func (a *Api) Get(url string) (*[]byte, error) {
        res, err := a.RawGet(url, nil)
 
-       log := httpLogger(res.Request)
+       log := logrus.WithFields(logrus.Fields{"url": url, "method": "GET"})
        if err != nil {
                log.Debugf("Network Error: %s", err)
                return nil, err
~~~

Another might be to make `httpLogger(url, method string)` and update the call sites. Not sure which you prefer. I re-recreate this by updating /etc/hosts to some bogus IP for api.foundries.io.

**Comment by vkhoroz** [comment]
It looks like this was fixed by commit b3389195.
Other places seems sane.
